bvm/ballerina-core/src/main/java/org/ballerinalang/model/types/BArrayType.java

```diff
@@ -99,19 +94,19 @@ public BType getElementType() {
         int tag = elementType.getTag();
         switch (tag) {
```

Comment:
same as above

Reply:
We cannot fully remove the switch case because there is a special case for DecimalArray and also Ref Array. But we could refine the switch a bit.